makefiles/multislot.mk

```diff
@@ -0,0 +1,42 @@
+ifneq (,$(filter mcuboot, $(MAKECMDGOALS)))
+IMGTOOL ?= $(RIOTBASE)/dist/tools/mcuboot/imgtool.py
+override IMGTOOL := $(abspath $(IMGTOOL))
+
+ifndef SLOT0_SIZE
+$(error Board $(BOARD) does not define multislot parameters!)
+else
+export FLASH_OFFSET := $(SLOT0_SIZE)
+endif
+
+export BINFILE ?= $(BINDIR)/$(APPLICATION).bin
+export SIGN_BINFILE = $(BINDIR)/signed-$(APPLICATION).bin
+export OFLAGS = -O binary
+
+create-key: key.pem
+
+key.pem:
+	$(Q)$(IMGTOOL) keygen -k key.pem -t rsa-2048
+
+mcuboot: create-key all
+	@$(COLOR_ECHO)
+	@$(COLOR_ECHO) '${COLOR_PURPLE}Re-linking for MCUBoot at $(SLOT0_SIZE)...${COLOR_RESET}'
+	@$(COLOR_ECHO)
+	$(Q)$(_LINK) $(LINKFLAGPREFIX)--defsym=offset="$(SLOT0_SIZE)" \
```

Comment:
here you can use ```--defsym=offset=$$(($(SLOT0_SIZE) - $(IMAGE_HDR))"``` and ```--defsym=length="$$(($(SLOT1_SIZE) - $(IMAGE_HDR)"```. IMAGE_HDR should also be renamed to IMAGE_HDR_SIZE.
```--defsym=image_header...``` can then be dropped, along with the linker script addition for img_hdr.